lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1229,7 +1253,23 @@ func (d *zfs) RefreshVolume(vol Volume, srcVol Volume, srcSnapshots []Volume, al
 
 // DeleteVolume deletes a volume of the storage device. If any snapshots of the volume remain then
 // this function will return an error.
+// For image volumes, both filesystem and block volumes will be removed.
 func (d *zfs) DeleteVolume(vol Volume, op *operations.Operation) error {
+	for _, suffix := range zfsAllowedFilesystems {
```

Comment:
This bit isn't needed if we don't encode the filesystem into the image volume name.